content/200-concepts/100-components/02-prisma-client/260-client-extensions/120-query.mdx

```diff
@@ -1,16 +1,176 @@
 ---
 title: 'Create custom Prisma Client queries'
-metaTitle: 'Prisma Client extensions (Preview)'
+metaTitle: 'Prisma Client extensions: query component (Preview)'
 metaDescription: 'Extend the functionality of Prisma Client'
 tocDepth: 3
 ---
 
 <TopBlock>
 
-<Admonition type="warning">
+From version 4.7.0, you can use the `query` [Prisma Client extensions](/concepts/components/prisma-client/client-extensions) component to hook into the query life-cycle and modify an incoming query or its result.
 
-We plan to add `query` extensions to a future version of Prisma. This page is a placeholder.
-
-</Admonition>
+You can use the `query` extension component as an alternative to [middlewares](/concepts/components/prisma-client/middleware#performance-and-appropriate-use-cases). Unlike middlewares, the `query` extension component gives you [user isolation](/concepts/components/prisma-client/client-extensions#extended-clients) and end-to-end type safety. [Learn more](#query-extensions-versus-middlewares).
```

Comment:
Q: What is user isolation?

Reply:
I think what we mean here is a row-level security scenario. Maybe I should say that - something like:
```suggestion
You can use the `query` extension component as an alternative to [middlewares](/concepts/components/prisma-client/middleware#performance-and-appropriate-use-cases). Unlike middlewares, the `query` extension component gives you [user isolation](/concepts/components/prisma-client/client-extensions#extended-clients), such as in a row-level security (RLS) scenario, and end-to-end type safety. [Learn more](#query-extensions-versus-middlewares).
```

Reply:
@millsp  What do you think of the above?